libsolidity/interface/CompilerStack.h

```diff
@@ -103,6 +103,9 @@ class CompilerStack: boost::noncopyable
 	/// Sets the given source code as the only source unit apart from standard sources and parses it.
 	/// @returns false on error.
 	bool parse(std::string const& _sourceCode);
+	bool analyze();
```

Comment:
i added an extra check at the beginning of the analyze function to see whether the m_sources and m_sourceUnitsByName are empty. 
If so, should it throw a BOOSTEXCEOPTION or should i use solAssert?